napari/_qt/dialogs/qt_plugin_dialog.py

```diff
@@ -998,7 +1000,7 @@ def _install_packages(self, packages: Sequence[str] = ()):
     def _handle_yield(self, data: Tuple[PackageMetadata, bool]):
         project_info, is_available = data
         if project_info.name in self.already_installed:
-            self.installed_list.tag_outdated(project_info)
+            self.installed_list.tag_outdated(project_info, is_available)
```

Comment:
As I inspect `hub_plugin_info` function from `napari.plugins.hub` its only check if most recent version is available on conda-forge. 

But if the user has a package in version `0.1` and versions `0.2` and `0.3` are published. But `0.3` does not reach yet conda -forge then he will not see a button with information that an update is available. But should. 

This also does not solve the problems of the package that the most recent version requires never dependency than it is pinned as core napari dependecy.

Reply:
>But if the user has a package in version 0.1 and versions 0.2 and 0.3 are published. But 0.3 does not reach yet conda -forge then he will not see a button with information that an update is available. But should.

Yes that is correct, but to solve this we need more information from the hub. So in the meantime is the best solution we can offer.

>This also does not solve the problems of the package that the most recent version requires never dependency than it is pinned as core napari dependecy.

I do not understand what you mean. Could you elaborate? thanks!

Reply:
> I do not understand what you mean. Could you elaborate? thanks!

Simple example. Napari bundle pins `superqt` to `0.3.2` In the meantime superqt `0.3.3` is released. Someone decides that he needs a feature from `0.3.3` and sets it as a minimum dependency. Create release. upload to pypi and anaconda. 

The current code will report that there is an update, but every installation will fail without explanation. 

> Yes that is correct, but to solve this we need more information from the hub (which currently only lists the latest from pypi). So in the meantime is the best solution we can offer.

No, you depend also on information from conda so you could calculate maximum available version from here: 

https://github.com/napari/napari/blob/3b3e2d1d9c63da141144e500b838ee9ea4b585cb/napari/plugins/hub.py#L56-L59

Reply:
>The current code will report that there is an update, but every installation will fail without explanation.

Yes but that is a separate issue, please open one :)

Reply:
>No, you depend also on information from conda so you could calculate maximum available version from here:

You are right, added a fix!